src/fps/proposal/MultisigProposal.sol

```diff
@@ -0,0 +1,854 @@
+    EnumerableSet.AddressSet private _allowedStorageAccesses;
+
+    /// @notice Struct to store information about an action
+    /// @param target The address of the target contract
+    /// @param value The amount of ETH to send with the action
+    /// @param arguments The calldata to send with the action
+    /// @param description A description of the action
+    struct Action {
+        address target;
+        uint256 value;
+        bytes arguments;
+        string description;
+    }
+
+    /// @notice Struct to store information about a token/Eth transfer
+    /// @param to The address of the recipient
+    /// @param value The amount of tokens/Eth to transfer
+    /// @param tokenAddress The address of the token contract
+    struct TransferInfo {
+        address to;
+        uint256 value;
+        address tokenAddress;
+    }
+
+    /// @notice Struct to store information about a state change
+    /// @param slot The storage slot that is being updated
+    /// @param oldValue The old value of the storage slot
+    /// @param newValue The new value of the storage slot
+    struct StateInfo {
+        bytes32 slot;
+        bytes32 oldValue;
+        bytes32 newValue;
+    }
+
+    /// @notice Struct to store information about run flags
+    /// @param doMock Flag to determine if the proposal should be mocked
+    /// @param doBuild Flag to determine if the proposal should be built
+    /// @param doSimulate Flag to determine if the proposal should be simulated
+    /// @param doValidate Flag to determine if the proposal should be validated
+    /// @param doPrint Flag to determine if the proposal should be printed
+    struct RunFlags {
+        bool doMock;
+        bool doBuild;
+        bool doSimulate;
+        bool doValidate;
+        bool doPrint;
+    }
+
+    /// @notice transfers during proposal execution
+    mapping(address => TransferInfo[]) private _proposalTransfers;
+
+    /// @notice state changes during proposal execution
+    mapping(address => StateInfo[]) private _stateInfos;
+
+    /// @notice addresses involved in state changes or token transfers
+    EnumerableSet.AddressSet private _proposalTransferFromAddresses;
+
+    /// @notice addresses whose state is updated in proposal execution
+    EnumerableSet.AddressSet internal _proposalStateChangeAddresses;
+
+    /// @notice stores the gnosis safe accesses for the proposal
+    VmSafe.StorageAccess[] internal _accountAccesses;
+
+    /// @notice starting snapshot of the contract state before the calls are made
+    uint256 private _startSnapshot;
+
+    /// @notice list of actions to be executed, regardless of proposal type
+    /// they all follow the same structure
+    Action[] public actions;
+
+    /// @notice proposal name, e.g. "OIP15".
+    /// @dev set in the proposal config file
+    string public override name;
+
+    /// @notice proposal description.
+    /// @dev set in the proposal config file
+    string public override description;
```

Comment:
Similar comment as before, our tasks aren't stored on chain and our docs should contain all the info/context a dev needs to understand the scope of an upgrade.